Sample Applications/WPFGallery/ViewModels/MainWindowViewModel.cs

```diff
@@ -302,4 +304,9 @@ private bool FindNavigationItemsHierarchyFromPageType(Type pageType, ICollection
 
         return false;
     }
+
+    internal void UpdateNavigationBackStatus()
+    {
+        NavigationServiceCanGoBack = _navigationService.CanGoBack();  
```

Comment:
Instead of NavigationServiceCanGoBack, it would be better to use the name, CanNavigateBack
Following on that,  we can replace the name CanGoBack with something like IsBackHistoryNonEmpty.
Similarly, the name UpdateNavigationBackStatus can be updated to UpdateCanNavigateBack.

These are just suggestions.

Reply:
Thanks for the suggestions. I made this name in sync with default names used in NavigationService. It is better to use the same naming convention over the framework and the sample app.  

For ref: https://learn.microsoft.com/en-us/dotnet/api/system.windows.navigation.navigationservice.cangoback?view=windowsdesktop-8.0

Reply:
I get your point, but this also causes confusion right, here we have designed our own NavigationService and the method names when kept similar may lead to developers thinking that we are using the other service.

Moreover, NavigationServiceCanGoBack is redundant here.

Reply:
Ok will update the name as suggested.